### PR TITLE
[API Compatibility No.187] Add target alias for F.multi_margin_loss -part

### PR DESCRIPTION
**Pull request by Aidenwu0209**
### PR Category
User Experience

### PR Types
New features

### Description
Complete API Compatibility No.187 in issue #76301.

- Add kwargs alias support for `paddle.nn.functional.multi_margin_loss`: `target -> label` via `@param_one_alias(["label", "target"])`.
- Keep signature, default values, and numerical logic unchanged.
- Update docstring of `label` parameter to mark alias `target`.
- Add legacy test case to verify `target=` and `label=` give identical outputs in dygraph mode for `reduction="none"` and `reduction="mean"`.

Issue: https://github.com/PaddlePaddle/Paddle/issues/76301

### 是否引起精度变化
否

### CODE REVIEW

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Comment by zhwesky2010** [comment]
loss这一系列API，还有一个整体的事情需要干，torch比paddle要多size_average、reduce，目前是通过legacy_reduction_decorator/legacy_reduction_special_decorator来进行报错增强的，但是还是无法根本解决问题，也就是无法通过paconvert测试。

需要增强这个装饰器，针对size_average、reduce直接设置为paddle的reduce参数。无需报错。然后需要在paconvert上增加size_average、reduce的测试。

这个单独再提PR吧，本PR可先合入。

**Comment by luotao1** [comment]
可以考虑merge develop，[CI-H / Coverage build (pull_request)](https://github.com/PaddlePaddle/Paddle/actions/runs/22656149424/job/65811759989?pr=78134) 一直过不了

File: test/legacy_test/test_multimarginloss.py [hunk 1]
```diff
@@ -329,6 +329,36 @@ def test_MultiMarginLoss_dimension(self):
         )
         paddle.enable_static()
 
+    def test_MultiMarginLoss_target_alias(self):
+        paddle.disable_static()
+        try:
```
Comment (zhwesky2010) [inline test/legacy_test/test_multimarginloss.py]: 测试代码不要使用try（除非是专门测异常抛出的），需要保证测试case正确可运行

Comment (zhwesky2010) [inline test/legacy_test/test_multimarginloss.py]: 测试代码不要使用try

Comment (zhwesky2010) [inline test/legacy_test/test_multimarginloss.py]: 测试代码不要使用try

**Review by zhwesky2010 — CHANGES_REQUESTED** [review]

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

**Comment by luotao1** [comment]
@Aidenwu0209 可以merge下develop，看看能否通过 CI-H 流水线

**Review by zhwesky2010 — APPROVED** [review]
